### PR TITLE
Removed csv, switched to using arrays

### PR DESCRIPTION
**Pull request by shaharyarn**
Linked marine-python PR:
https://github.com/tomlegkov/marine-python/pull/35

### CODE REVIEW

File: marine.c [hunk 1]
```diff
@@ -572,17 +561,12 @@ WS_DLL_PUBLIC marine_result *marine_dissect_packet(int filter_id, unsigned char
...
-        if (passed) {
-            result->result = 1;
-            result->output = output;
-        } else {
-            if (output != NULL) {
-                free(output);
-            }
-            result->result = 0;
-        }
+        unsigned int output_len = filter->output_fields == NULL ? 0 : filter->output_fields->fields->len;
```
Comment (yehonatanz) [inline marine.c:566]: Why the extra variable?

File: marine.c [hunk 2]
```diff
@@ -362,39 +362,28 @@ marine_write_specified_fields(packet_filter *filter, epan_dissect_t *edt, char *
...
-            if (fields->quote != '\0') {
-                output[counter++] = fields->quote;
-            }
+            output[i] = (char *) g_malloc0(4096);
+            int counter = 0;
 
-            /* Output the array of (partial) field values */
             for (j = 0; j < g_ptr_array_len(fv_p); j++) {
                 str = (gchar *) g_ptr_array_index(fv_p, j);
                 for (char *p = str; *p != '\0'; p++) {
```
Comment (yehonatanz) [inline marine.c:376]: At least limit this loop to 4095 iterations or something - better slicing the output than overflowing.
Have you tried figuring out the length properly? How does wireshark/tshark do it?

**Review by yehonatanz — CHANGES_REQUESTED** [review]
Looks good, but I had some notes.

File: marine.c (hunk 2 above)
Comment (tomlegkov) [reply marine.c:376]: I agree with @yehonatanz. If you can't figure out how Wireshark does it, `realloc` or limiting the loop should suffice.

**Review by tomlegkov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**